tests/sklearn/test_sklearn_models.py

```diff
@@ -1871,3 +1917,91 @@ def test_rounding_consistency_for_regular_models(
         metric,
         is_weekly_option,
     )
+
+
+@pytest.mark.parametrize("model_class, parameters", get_sklearn_tree_models_and_datasets())
+@pytest.mark.parametrize("n_bits", [2, 5, 10])
+def test_fhe_sum_for_tree_based_models(
+    model_class,
+    parameters,
+    n_bits,
+    load_data,
+    is_weekly_option,
+    verbose=True,
+):
+    """Test that the tree ensembles' output are the same with and without the sum in FHE."""
+
+    if verbose:
+        print("Run check_fhe_sum_for_tree_based_models")
+
+    model = instantiate_model_generic(model_class, n_bits=n_bits)
+
+    x, y = get_dataset(model_class, parameters, n_bits, load_data, is_weekly_option)
+
+    predict_method = (
+        model.predict_proba if is_classifier_or_partial_classifier(model) else model.predict
+    )
+
+    check_fhe_sum_for_tree_based_models(
+        model,
+        x,
+        y,
+        predict_method,
+        is_weekly_option,
+    )
+
+
+# This test should be extended to all built-in models.
+# FIXME: https://github.com/zama-ai/concrete-ml-internal#4234
```

Comment:
I think you wanted to put `https://github.com/zama-ai/concrete-ml-internal/issues/4234`